### ISSUE TITLE
Scores/Rating not stored Round II

### ISSUE THREAD

**Issue by micheljung** [FAForever/server#227]
Just played a game:

```
+--------------+---------+-----------+-------+------------+-----------------+
| login        | gameid  | scoreTime | score | after_mean | after_deviation |
+--------------+---------+-----------+-------+------------+-----------------+
| Boofy        | 5268903 | NULL      |    -1 |       NULL |            NULL |
| hm5555       | 5268903 | NULL      |    -1 |       NULL |            NULL |
| Pickral      | 5268903 | NULL      |    -1 |       NULL |            NULL |
| Lokriel      | 5268903 | NULL      |    -1 |       NULL |            NULL |
| Downlord     | 5268903 | NULL      |    -1 |       NULL |            NULL |
| IvoFromBG    | 5268903 | NULL      |    -1 |       NULL |            NULL |
| DarthRaven   | 5268903 | NULL      |    -1 |       NULL |            NULL |
| FAFgaruny    | 5268903 | NULL      |    -1 |       NULL |            NULL |
| saneyka88    | 5268903 | NULL      |    -1 |       NULL |            NULL |
| XTEM         | 5268903 | NULL      |    -1 |       NULL |            NULL |
| WickedHermit | 5268903 | NULL      |    -1 |       NULL |            NULL |
| Mechanical   | 5268903 | NULL      |    -1 |       NULL |            NULL |
+--------------+---------+-----------+-------+------------+-----------------+
```

In the server log I see that scores have been reported:

```
2016-10-13 02:03:32.734
pam_unix(sshd:session): session closed for user downlord
2016-10-13 01:54:21.566
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 6: victory 10
2016-10-13 01:54:21.566
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 5: victory 10
2016-10-13 01:54:21.566
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 4: victory 10
2016-10-13 01:54:21.566
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 3: victory 10
2016-10-13 01:54:21.565
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 2: victory 10
2016-10-13 01:54:21.535
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 1: victory 10
2016-10-13 01:54:13.527
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 9: defeat -10
2016-10-13 01:54:09.575
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 1: score 2
2016-10-13 01:52:45.722
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 8: defeat -10
2016-10-13 01:50:44.867
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 10: defeat -10
2016-10-13 01:50:23.239
Player(Downlord, 21447, (1277.107188164776, 84.77044339666246), (1060.36, 57.0235)) reported result for army 1: score 1
```

The game.log shows `GAMERESULT`, but the JsonStats (used for achievements) are missing:

```
INFO: playable area coordinates are { 4, 200, 1016, 820 }
INFO: Offmapareas are {
INFO:   { x0=-96, x1=1116, y0=100, y1=200 },
INFO:   { x0=-96, x1=1116, y0=820, y1=920 },
INFO:   { x0=-96, x1=4, y0=200, y1=820 },
INFO:   { x0=1016, x1=1116, y0=200, y1=820 }
INFO: }
INFO: Hooked /lua/ui/game/scoreaccum.lua with /mods/supremeeconomy/hook/lua/ui/game/scoreaccum.lua
INFO: Hooked /lua/defaultantiprojectile.lua with /schook/lua/defaultantiprojectile.lua
INFO: Hooked /lua/ui/game/commandgraph.lua with /schook/lua/ui/game/commandgraph.lua
INFO: "AdvancedEngineering"
INFO: "ResourceAllocation"
WARNING: Error running OnStopBuild script in Entity urb0103 at 2a33cf08: ...\gamedata\units.scd\units\urb0103\urb0103_script.lua(37): attempt to call method `SetGoal' (a nil value)
         stack traceback:
            ...\gamedata\units.scd\units\urb0103\urb0103_script.lua(37): in function `StopArmsMoving'
            ...mdata\faforever\gamedata\lua.nx2\lua\cybranunits.lua(273): in function <...mdata\faforever\gamedata\lua.nx2\lua\cybranunits.lua:270>
INFO: GAMERESULT : \000defeat -10
INFO: "T3Engineering"
INFO: Strat Icon Mod Error: updated strat icon required for: \000icon_bot3_armored
INFO: Strat Icon Mod Error: updated strat icon required for: \000icon_bot3_antiair
INFO: GAMERESULT : \000defeat -10
INFO: GAMERESULT : \000defeat -10
INFO: CWldSession::DoBeat() unknown entity id (0x2ff00628) supplied in a pose update.
INFO: CWldSession::DoBeat() unknown entity id (0x2ff00a27) supplied in a pose update.
WARNING: ACU kill detected. Rating for ranked games is now enforced.
INFO: GAMERESULT : \000score 1
INFO: GAMERESULT : \000defeat -10
INFO: GAMERESULT : \000defeat -10
INFO: CWldSession::DoBeat() unknown entity id (0x2ff00e6e) supplied in a pose update.
INFO: CWldSession::DoBeat() unknown entity id (0x2ff00ea6) supplied in a pose update.
WARNING: ACU kill detected. Rating for ranked games is now enforced.
INFO: GAMERESULT : \000score 2
INFO: GAMERESULT : \000defeat -10
INFO: GAMERESULT : \000victory 10
INFO: GAMERESULT : \000victory 10
INFO: GAMERESULT : \000victory 10
INFO: GAMERESULT : \000victory 10
INFO: GAMERESULT : \000victory 10
INFO: GAMERESULT : \000victory 10
INFO: syncing max ping markers: \00015
INFO: Minimized true
WARNING: GetResource: Invalid name ""
INFO: Can't find texture "" -- trying fallback.
WARNING: GetResource: Invalid name ""
INFO: Can't find texture "" -- trying fallback.
WARNING: GetResource: Invalid name ""
INFO: Can't find texture "" -- trying fallback.
WARNING: GetResource: Invalid name ""
INFO: Can't find texture "" -- trying fallback.
WARNING: GetResource: Invalid name ""
INFO: Can't find texture "" -- trying fallback.
INFO: Minimized false
```

**Comment by Crotalus** [comment]
Could this also be due to #229?

**Comment by micheljung** [comment]
most likely, yes.

**Comment by micheljung** [comment]
Assuming that this has been solved with #229
